sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/EventProcessorClientEventSource.cs

```diff
@@ -188,5 +190,82 @@ protected EventProcessorClientEventSource() : base(EventSourceName)
                 WriteEvent(26, identifier ?? string.Empty, eventHubName ?? string.Empty, consumerGroup ?? string.Empty, errorMessage ?? string.Empty);
             }
         }
+
+        /// <summary>
+        /// Writes an event with four string arguments into a stack allocated <see cref="EventData"/> struct to avoid
+        /// the parameter array allocation on the WriteEvent methods.
+        /// </summary>
+        /// <param name="eventId">The event identifier.</param>
+        /// <param name="arg1">The first argument.</param>
+        /// <param name="arg2">The second argument.</param>
+        /// <param name="arg3">The third argument.</param>
+        /// <param name="arg4">The fourth argument.</param>
+        [NonEvent]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe void WriteEvent(int eventId, string arg1, string arg2, string arg3, string arg4)
```

Comment:
I'm surprised.... I remember this set of logs being much more extensive.   I guess they mostly got attached to the base class and are in the main source now.   Yay!